blog post for team Recipe building and sharing (#2495)
block/goose: documentation/blog/2025-05-09-developers-ai-playbook-for-team-efficiency/index.md

---

> title: "Championship Driven Development: Your Team's AI Playbook for Peak Performance"

Reviewer: cool title

> description: How AI-powered 'plays' can transform your dev team into a high-scoring sports team, streamlining game plans for debugging, changelogs, PRs, and more.

Reviewer: not for this blog post but in the future , i think the description could be more concise. dont even really need the and more part

authors:
    - ian
---

![blog cover](cdd-playbook.png)

# A Developer's Playbook: Using AI for Team Efficiency

> Development teams can operate like sports teams. Each member has a role and a shared playbook helps coordinate efforts. Let's explore how AI-driven "plays" can form a starter "playbook" for your dev team, helping with common technical tasks. You can use these "recipes" with [Goose](/) to leverage the Model Context Protocol (MCP) to make you more productive.

Reviewer: ```suggestion
Development teams can operate like sports teams. Each member has a role and a shared playbook helps coordinate efforts. Let's explore how AI-driven "plays" can form a starter "playbook" for your dev team, helping with common technical tasks. You can use these "recipes" with [Goose](/) to leverage the [Model Context Protocol (MCP)](https://modelcontextprotocol.io/) to make you more productive.
```

> Development teams can operate like sports teams. Each member has a role and a shared playbook helps coordinate efforts. Let's explore how AI-driven "plays" can form a starter "playbook" for your dev team, helping with common technical tasks. You can use these "recipes" with [Goose](/) to leverage the Model Context Protocol (MCP) to make you more productive.

Reviewer: Maybe take recipes out of quotes because it's not the fake name..it's the real name?

* New developers need to learn the team’s processes and tools. This takes time.
* Ensuring consistent quality across all work requires clear standards, e.g.: a sports team practicing plays over and over to achieve consistent execution.
* Teams often use many tools, from IDEs and version control to CI/CD pipelines and issue trackers.
* Managing these tools and the workflows between them can be complex.

## Benefits of Using an AI Playbook

Using a shared AI playbook provides several benefits for a development team:
* **Faster Onboarding:** New team members can use existing recipes to learn standard procedures and become productive more quickly.
* **Improved Consistency:** Standardized recipes ensure tasks are performed the same way every time, leading to more predictable results.
* **Increased Efficiency:** Automating routine tasks frees developers to focus on more complex problem-solving.
* **Knowledge Sharing:** Recipes can codify team knowledge and best practices, making them accessible to everyone.

As teams adopt AI tools like Goose, the ability to define and share these automated workflows will become increasingly important.

## AI Plays: Standardizing Your Team's Workflows

> Goose can help standardize and automate these tasks, by [creating recipes](/docs/guides/session-recipes). As a developer on your team uses Goose, they can create a recipe that describes how to perform a task, and then share that with the rest of the team. These recipies can be shared and reused, and improved over time, just like a sports team’s playbook.

Reviewer: ```suggestion
Goose can help standardize and automate these tasks, by [creating recipes](/docs/guides/session-recipes). As a developer on your team uses Goose, they can create a recipe that describes how to perform a task, and then share that with the rest of the team. These recipes can be shared and reused, and improved over time, just like a sports team’s playbook.
```

Recipes are built with an understanding of the workflow you want Goose to help with, and these may involve one or more MCP servers, such as [GitHub](/docs/tutorials/github-mcp/) or [PostgreSQL](/docs/tutorials/postgres-mcp/). The recipes are designed to be reusable and adaptable, allowing developers to create a library that can be used across different projects.

A shared playbook of AI plays helps everyone on the team perform tasks consistently. It can also reduce the time spent on repetitive work.

## Goose Recipes: The Building Blocks of Your Playbook

For a kitchen-related analogy as an overview, check out [Rizel's](/blog/authors/rizel/) recent blog post, [A Recipe for Success](/blog/2025/05/06/recipe-for-success).

> A Goose Recipe can be saved from a current Goose session, or written as a YAML file from scratch. It includes instructions for the AI to follow, a prompt to generate the AI response, and can also include parameters with data types, and which extensions are needed to use the recipe.

Reviewer: ```suggestion
A Goose Recipe can be saved from a current Goose session, or written as a YAML file from scratch. IIt includes instructions for the AI to follow, a prompt for the AI response, optional parameters with data types, and a list of required extensions.
```

  - Based on actual code changes
  - No body text needed for straightforward changes

  Do not:
  - Ask for confirmation or additional input
  - Create placeholder content
  - Include TODO items
  - Add WIP markers
  {% endif %}
```

</details>

### Other Potential Plays for Your Playbook

Your team can create plays for many other tasks:
* **Debugging Assistance:** A play could guide a developer or an AI through initial steps for diagnosing common issues, by checking specific logs or running predefined commands.
* **Log Analysis:** An AI play can define a standard procedure for querying and summarizing log data to identify problems.
* **Documentation Updates:** A "Readme Bot" could have AI assist in generating or updating project README files.
* **Content Migration:** The "dev guide migration" recipe could provide a structured approach to migrating documentation content, ensuring information is preserved and correctly formatted.

## What kinds of tasks can your team automate?

We'd love for you to share your ideas with us! You can do this by creating a recipe and sharing it with the [Goose community on Discord](http://discord.gg/block-opensource).

<head>

>   <meta property="og:title" content="A Developer's Playbook: Using AI for Team Efficiency" />

Reviewer: ```suggestion
  <meta property="og:title" content="Championship Driven Development: Your Team's AI Playbook for Peak Performance" />
```

  <meta property="og:type" content="article" />
  <meta property="og:url" content="https://block.github.io/goose/blog/2025/05/09/developers-ai-playbook-for-team-efficiency" />
  <meta property="og:description" content="Learn how AI-driven 'plays,' based on Model Context Protocol, can help development teams improve common workflows like changelog generation and pull requests." />
  <meta property="og:image" content="TODO" />
  <meta name="twitter:card" content="summary_large_image" />
  <meta property="twitter:domain" content="block.github.io/goose" />

>   <meta name="twitter:title" content="A Developer's Playbook: Using AI for Team Efficiency" />

Reviewer: ```suggestion
  <meta name="twitter:title" content="Championship Driven Development: Your Team's AI Playbook for Peak Performance" />
```
